### PR TITLE
Update ListAmount function.

### LINKED ISSUE #155: Update ListAmount function in the expenseSplitter.

### PR DESCRIPTION
**Pull request by sheexiong**

### CODE REVIEW

**Comment by sheexiong** [comment]
@ailanthustng I realise the `additem` command got some `nullpointer exception error`, maybe you could debug it after merging this PR.

File: src/main/java/seedu/eylah/expensesplitter/logic/commands/AddItemCommand.java [hunk 1]
```diff
@@ -52,15 +52,10 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         model.addEntry(toBeAdded);
         for (Person person : persons) {
-            if (model.hasPerson(person)) {
-                model.addAmount(person, amount);
-                // person.addAmount(amount); //not sure if necessary
```
Comment (WillySeahh) [inline src/main/java/seedu/eylah/expensesplitter/logic/commands/AddItemCommand.java:57]: This file looks good!

File: src/main/java/seedu/eylah/expensesplitter/model/PersonAmountBook.java [hunk 2]
```diff
@@ -110,8 +109,20 @@ public void removePerson(Person key) {
 
     @Override
     public String toString() {
-        return persons.asUnmodifiableObservableList().size() + " persons";
-        // TODO: refine later
+        final StringBuilder builder = new StringBuilder();
+        ObservableList<Person> personsList = persons.asUnmodifiableObservableList();
+        builder.append("Person : Amount\n");
+        for (Person person : personsList) {
+            int index = personsList.indexOf(person) + 1;
```
Comment (WillySeahh) [inline src/main/java/seedu/eylah/expensesplitter/model/PersonAmountBook.java:116]: Nice! If i understand your code correctly, this basically adds the Json parameter (amount) into personamount.json right?

**Review by WillySeahh — DISMISSED** [review]
LGTM in general. Remember to do a release by 2359 today

File: src/main/java/seedu/eylah/expensesplitter/model/PersonAmountBook.java (hunk 2 above)
Comment (sheexiong) [reply src/main/java/seedu/eylah/expensesplitter/model/PersonAmountBook.java:116]: For that part, I add the field in the `JsonAdaptedPerson` class.

File: src/main/java/seedu/eylah/expensesplitter/logic/commands/AddItemCommand.java [hunk 3]
```diff
@@ -52,15 +52,10 @@ public CommandResult execute(Model model) throws CommandException {
...
-            } else {
-                // add person, then add amount to that person
+            if (!model.hasPerson(person)) {
                 model.addPerson(person);
-                model.addAmount(person, amount);
-                // person.addAmount(amount); //not sure if necessary
             }
+            model.addAmount(person, amount);
```
Comment (ailanthustng) [inline src/main/java/seedu/eylah/expensesplitter/logic/commands/AddItemCommand.java:58]: Alright! This looks better. Thanks @sheexiong !

**Review by WillySeahh — APPROVED** [review]
LGTM

File: src/main/java/seedu/eylah/expensesplitter/model/PersonAmountBook.java [hunk 4]
```diff
@@ -110,8 +109,20 @@ public void removePerson(Person key) {
 
     @Override
     public String toString() {
```
Comment (ailanthustng) [inline src/main/java/seedu/eylah/expensesplitter/model/PersonAmountBook.java:111]: Looks good!

**Review by ailanthustng — APPROVED** [review]
LGTM

**Pull request merged**